src/entity/schema.ts

```diff
@@ -0,0 +1,389 @@
+        colors: [ CardColor.RED, CardColor.GREEN, CardColor.BLUE, CardColor.YELLOW ] as readonly CardColor[],
+        name: 'reset'
+    },
+    RED_INVISIBLE: {
+        type: CardType.INVISIBLE,
+        colors: [ CardColor.RED ] as readonly CardColor[],
+        name: 'red invisible'
+    },
+    GREEN_INVISIBLE: {
+        type: CardType.INVISIBLE,
+        colors: [ CardColor.GREEN ] as readonly CardColor[],
+        name: 'green invisible'
+    },
+    BLUE_INVISIBLE: {
+        type: CardType.INVISIBLE,
+        colors: [ CardColor.BLUE ] as readonly CardColor[],
+        name: 'blue invisible'
+    },
+    YELLOW_INVISIBLE: {
+        type: CardType.INVISIBLE,
+        colors: [ CardColor.YELLOW ] as readonly CardColor[],
+        name: 'yellow invisible'
+    },
+    RED_NOMINATE: {
+        type: CardType.NOMINATE,
+        colors: [ CardColor.RED ] as readonly CardColor[],
+        name: 'red nominate'
+    },
+    GREEN_NOMINATE: {
+        type: CardType.NOMINATE,
+        colors: [ CardColor.GREEN ] as readonly CardColor[],
+        name: 'green nominate'
+    },
+    BLUE_NOMINATE: {
+        type: CardType.NOMINATE,
+        colors: [ CardColor.BLUE ] as readonly CardColor[],
+        name: 'blue nominate'
+    },
+    YELLOW_NOMINATE: {
+
+        type: CardType.NOMINATE,
+        colors: [ CardColor.YELLOW ] as readonly CardColor[],
+        name: 'yellow nominate'
+    },
+    MULTI_NOMINATE: {
+        type: CardType.NOMINATE,
+        colors: [ CardColor.RED, CardColor.GREEN, CardColor.BLUE, CardColor.YELLOW ] as readonly CardColor[],
+        name: 'multi nominate'
+    }
+} as const;
+
+export type Card = ObjectValues<typeof Card>;
+
+export const CardSchema = z.object({
+    type: z.string(),
+    colors: z.array(z.nativeEnum(CardColor)),
+    name: z.string(),
+    value: z.number().int().positive().optional()
+});
+
+export type CardSchema = z.infer<typeof CardSchema>;
+
+// ### User ###
+export const UserSchema = z.object({
+    username: z.string()
+});
+
+export type UserSchema = z.infer<typeof UserSchema>;
+
+// ### Player ###
+export const PlayerSchema = UserSchema.extend({
+    socketId: z.string().optional(),
+    cardAmount: z.number().int().positive(),
+    cards: z.array(CardSchema).optional(),
+    ranking: z.number().int().optional(),
+    disqualified: z.boolean()
+});
+
+export type PlayerSchema = z.infer<typeof PlayerSchema>;
+
+export const PlayerInvitationSchema = PlayerSchema.extend({
+    accepted: z.boolean()
+});
+
+export type PlayerInvitationSchema = z.infer<typeof PlayerInvitationSchema>;
+
+// ### Action ###
+export enum ActionType {
+    TAKE = 'take',
+    DISCARD = 'discard',
+    NOMINATE = 'nominate',
+    NOPE = 'nope',
+    DISQUALIFIED = 'disqualified'
+}
+
+export const ActionSchema = z.object({
+    type: z.nativeEnum(ActionType),
+    explanation: z.string(),
+    player: PlayerSchema,
+    cardAmount: z.number().int().positive(),
+    cards: z.array(CardSchema).optional(),

```

Comment:
Maximale Anzahl sinnvoll?
```suggestion
    cards: z.array(CardSchema).optional().max(3),
```

Reply:
Gut dann wird das nicht übernommen :)